mikeio/dutil.py

```diff
@@ -95,22 +114,29 @@ def __init__(self, data, time, items):
                 f"Number of items in iteminfo {len(items)} doesn't match the data {n_items}."
             )
         self.data = data
-        self.time = time
+        self.time = pd.DatetimeIndex(time, freq='infer')
         self.items = items
 
     def __repr__(self):
         n_items = len(self.items)
 
         out = []
-        out.append("DataSet(data, time, items)")
-        out.append(f"Number of items: {n_items}")
-        out.append(f"Shape: {self.data[0].shape}")
-        out.append(f"{self.time[0]} - {self.time[-1]}")
+        out.append("<mikeio.DataSet>")
+        out.append(f"Dimensions: {self.data[0].shape}")
+        out.append(f"Time: {self.time[0]} - {self.time[-1]}")
+        if n_items > 10:
+            out.append(f"Number of items: {n_items}")
+        else:
+            out.append("Items:")
+            for item in self.items:
+                out.append(f"  {item}")
```

Comment:
could we add a numeration to this such that it would appear:

0: water level
1: u-velocity
2: v-velocity

for easier selection of items :-)

Reply:
Absolutely, convenient to select items by number for interactive use, although for non-interactive use I highly recommend to select items by name.
```
>>> ds = mikeio.read("tests/testdata/HD2D.dfsu")
>>> ds
<mikeio.DataSet>
Dimensions: (9, 884)
Time: 1985-08-06 07:00:00 - 1985-08-07 03:00:00
Items:
  0:  Surface elevation <Surface Elevation> (meter)
  1:  U velocity <u velocity component> (meter per sec)
  2:  V velocity <v velocity component> (meter per sec)
  3:  Current speed <Current Speed> (meter per sec)
>>> ds2 = ds[[0,3]]
>>> ds2
<mikeio.DataSet>
Dimensions: (9, 884)
Time: 1985-08-06 07:00:00 - 1985-08-07 03:00:00
Items:
  0:  Surface elevation <Surface Elevation> (meter)
  1:  Current speed <Current Speed> (meter per sec)
>>> ds2[["Surface elevation", "Current speed"]]
<mikeio.DataSet>
Dimensions: (9, 884)
Time: 1985-08-06 07:00:00 - 1985-08-07 03:00:00
Items:
  0:  Surface elevation <Surface Elevation> (meter)
  1:  Current speed <Current Speed> (meter per sec)
>>> ds3 = ds2[["Surface elevation", "Current speed"]]
>>> ds3
<mikeio.DataSet>
Dimensions: (9, 884)
Time: 1985-08-06 07:00:00 - 1985-08-07 03:00:00
Items:
  0:  Surface elevation <Surface Elevation> (meter)
  1:  Current speed <Current Speed> (meter per sec)
>>>
```